### PR TITLE
Adding Troubleshooting Info for GH007 Error during gh-gei Migration

### LINKED ISSUE #25280: Adding Troubleshooting Info for GH007 Error during gh-gei Migration
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer#troubleshooting-failed-migrations

### What part(s) of the article would you like to see updated?

Running a test migration with `gh-gei` revealed that I had activated the option "Block command line pushes that expose my email" in my profile email settings which produced the error `[ERROR] Git source migration failed` with the info `GH007: Your push would publish a private email address`. The solution was not directly obvious to me and it took some time and research that pointed me to that option. I was not aware that it was set in my profile. So it might help users to solve this error faster.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by aatmmr**
### Why:

Closes: #25280

### What's being changed (if available, include any code snippets, screenshots, or gifs):

A new section "`Git source migration failed` error" was added to the [Troubleshooting failed migrations](https://docs.github.com/en/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer#troubleshooting-failed-migrations) section.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@aatmmr Thank you for opening a PR and linking it to your issue! ✨ 

I'll get this triaged for review! :zap:

File: content/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ The log contains a record of each command you issued and all of the API requests
 - [`Subsystem 'sftp' could not be executed` error](#subsystem-sftp-could-not-be-executed-error)
 - [`Source export archive... does not exist` error](#source-export-archive-does-not-exist-error)
 - [`Repository rule violations found` error](#repository-rule-violations-found-error)
+- [`Git source migration failed` error](#git-source-migration-failed-error)
```
Comment (lecoursen) [inline content/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer.md:48]:
```suggestion
- [`Your push would publish a private email address` error](#your-push-would-publish-a-private-email-address-error)
```

File: content/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer.md [hunk 2]
```diff
@@ -118,6 +119,10 @@ If you receive a `Repository rule violations found` error, such as `GH013: Repos
 
 You can temporarily disable your rulesets during your migration, or you can use bypass mode or the bypass list to exempt your migration from configured rules. For more information, see "[AUTOTITLE]({% ifversion repo-rules-enterprise %}{% else %}/enterprise-cloud@latest{% endif %}/organizations/managing-organization-settings/managing-rulesets-for-repositories-in-your-organization)."
 
+### `Git source migration failed` error
```
Comment (lecoursen) [inline content/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer.md:122]:
```suggestion
### `Your push would publish a private email address` error
```

File: content/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer.md [hunk 3]
```diff
@@ -118,6 +119,10 @@ If you receive a `Repository rule violations found` error, such as `GH013: Repos
...
+
+If you receive a `Git source migration failed` error, such as `GH007: Your push would publish a private email address`, review the email settings of your account. Go to *Settings* > *Emails* and scroll to the checkbox "Block command line pushes that expose my email" and make sure it is unchecked.
```
Comment (lecoursen) [inline content/migrations/using-github-enterprise-importer/completing-your-migration-with-github-enterprise-importer/troubleshooting-your-migration-with-github-enterprise-importer.md:124]:
```suggestion
If you receive a `Git source migration failed` error with `GH007: Your push would publish a private email address`,  the Git source you're trying to migrate includes commits authored by an email address that you have blocked from being pushed to {% data variables.product.prodname_dotcom %}. For more information, see "[AUTOTITLE]({% ifversion not fpt or ghec %}/enterprise-cloud@latest{% endif %}/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/blocking-command-line-pushes-that-expose-your-personal-email-address){% ifversion fpt or ghec %}."{% else %}" in the  {% data variables.product.prodname_ghe_cloud %} documentation.{% endif %}

To resolve this error, you can either rewrite the Git history to remove the email address, or you can disable the "Block command line pushes that expose my email" setting. 
```

**Review by lecoursen — DISMISSED** [review]
Thanks @aatmmr! I'm going to apply some small formatting updates and then merge. ⚡

**Review by lecoursen — APPROVED** [review]

**Pull request merged**